pkg/webhook/admission/deployment/model_provisioning_injector_test.go

```diff
@@ -0,0 +1,265 @@
+						ObjectMeta: metav1.ObjectMeta{
+							Annotations: map[string]string{
+								constants.KFServiceModelProvisioningSourceURIAnnotationKey: "haha://foo",
+								constants.KFServiceModelProvisioningMountPathAnnotationKey: "/mnt",
+							},
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								v1.Container{
+									Name: "user-container",
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Annotations: map[string]string{
+								constants.KFServiceModelProvisioningSourceURIAnnotationKey: "haha://foo",
+								constants.KFServiceModelProvisioningMountPathAnnotationKey: "/mnt",
+							},
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								v1.Container{
+									Name: "user-container",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"ProvisionerInjected": {
+			original: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Annotations: map[string]string{
+								constants.KFServiceModelProvisioningSourceURIAnnotationKey: "gs://foo",
+								constants.KFServiceModelProvisioningMountPathAnnotationKey: "/mnt/somewhere",
+							},
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								v1.Container{
+									Name: "user-container",
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Annotations: map[string]string{
+								constants.KFServiceModelProvisioningSourceURIAnnotationKey: "gs://foo",
+								constants.KFServiceModelProvisioningMountPathAnnotationKey: "/mnt/somewhere",
+							},
+						},
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								v1.Container{
+									Name: "user-container",
+									VolumeMounts: []v1.VolumeMount{
+										{
+											Name:      "kfserving-provision-location",
+											MountPath: "/mnt/somewhere",
+											ReadOnly:  true,
+										},
+									},
+								},
+							},
+							InitContainers: []v1.Container{
+								v1.Container{
+									Name:  "model-provisioner",
+									Image: "kcorer/kfdownloader:latest",
+									Args:  []string{"gs://foo", "/mnt/somewhere"},
+									VolumeMounts: []v1.VolumeMount{
+										{
+											Name:      "kfserving-provision-location",
+											MountPath: "/mnt/somewhere",
+										},
+									},
+								},
+							},
+							Volumes: []v1.Volume{
+								v1.Volume{
+									Name: "kfserving-provision-location",
+									VolumeSource: v1.VolumeSource{
+										EmptyDir: &v1.EmptyDirVolumeSource{},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		"ProvisionerInjectedAndMountsPvc": {
```

Comment:
@hougangliu  check it out - pvc support implemented by adding the mounts and converting src path to be relative to the pvc mount..